In [ ]:
from math import factorial

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

def permutations(n):
    return int(factorial(n))

**1. Из колоды в 52 карты извлекаются случайным образом 4 карты.**

a) Найти вероятность того, что все карты – крести. 


Кол-во карт крести

$ \large{ \frac{52}{4} = 13 } $

In [11]:
a = combinations(13,4)
b = combinations(52,4)

print(f'Число сочетаний когда выбирается 4 из 13 крести: {a}')
print(f'Число сочетаний когда выбирается 4 карты из 52: {b}')
print(f'Вероятность, что все карты крести: {a/b}')

Число сочетаний когда выбирается 4 из 13 крести: 715
Число сочетаний когда выбирается 4 карты из 52: 270725
Вероятность, что все карты крести: 0.0026410564225690276


$ \large{ \frac{C_{13}^4}{C_{52}^4} = 0.00264 } $

б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

In [20]:
a = combinations(4,1)
b = combinations(48,3)
total_with_T = a*b
total = combinations(52,4)

print(f'Число сочетаний когда выбирается 1 туз из 4: {a}')
print(f'Число сочетаний когда выбирается 3 карты из 48 (исключаяя тузы): {b}')
print(f'Итоговое число сочетаний когда выбирается 4 карты и один из них туз: {total_with_T}')

print(f'Вероятность, что выберется хотябы 1 туз: {total_with_T/total}')

Число сочетаний когда выбирается 1 туз из 4: 4
Число сочетаний когда выбирается 3 карты из 48 (исключаяя тузы): 17296
Итоговое число сочетаний когда выбирается 4 карты и один из них туз: 69184
Вероятность, что выберется хотябы 1 туз: 0.25555083571890297


$ \large{ \frac{C_{4}^1*C_{48}^3}{C_{52}^4} = 0.25555 } $

**2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно.**

In [23]:
print(f'Вероятность: {(3/10)*(2/9)*(1/8)}')

Вероятность: 0.008333333333333333


$ \large{ P(3) = \frac{3}{10}*\frac{2}{9}*\frac{1}{8} = 0.00833 } $

**3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?**

In [21]:
print(f'Вероятность: {(9/15)*(8/14)*(7/13)}')

Вероятность: 0.18461538461538457


$ \large{ P(3) = \frac{9}{15}*\frac{8}{14}*\frac{7}{13} = 0.184615 } $

**4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?**

In [22]:
print(f'Вероятность: {(2/100)*(1/99)}')

Вероятность: 0.00020202020202020205


$ \large{ P(2) = \frac{2}{100}*\frac{1}{99} = 0.000202 } $